petri/logview_new/src/log_viewer.tsx

```diff
@@ -0,0 +1,432 @@
+    }, [pendingScrollIndex]);
+
+    return (
+        <div className="common-page-display">
+            <div className="common-page-header">
+                <LogViewerHeader
+                    runId={runId || 'unknown'}
+                    architecture={architecture}
+                    testNameRemainder={testName}
+                    fullTestName={fullTestName}
+                    searchFilter={searchFilter}
+                    setSearchFilter={setSearchFilter}
+                    searchInputRef={searchInputRef}
+                />
+            </div>
+
+            <div ref={logContainerRef} style={{ fontFamily: 'monospace', fontSize: '14px', position: 'relative' }}>
+                {filteredLogs.length === 0 && isSuccess ? (
+                    <div className="common-no-data">
+                        No log entries found
+                    </div>
+                ) : (
+                    <VirtualizedTable<LogEntry>
+                        data={filteredLogs}
+                        columns={columns}
+                        sorting={sorting}
+                        onSortingChange={setSorting}
+                        columnWidthMap={columnWidthMap}
+                        estimatedRowHeight={50}
+                        getRowClassName={(row) => {
+                            const logId = `log-${row.original.index}`;
+                            const isSelected = selectedRow === logId;
+                            const severityClass = `severity-${row.original.severity}`;
+                            return `${severityClass} ${isSelected ? 'selected' : ''}`;
+                        }}
+                        onRowClick={(row, event) => {
+                            const logId = `log-${row.original.index}`;
+                            handleRowClick(
+                                row.original.index,
+                                logId,
+                                event,
+                                selectedRow,
+                                setSelectedRow,
+                                location,
+                                navigate
+                            );
+                        }}
+                        scrollToIndex={pendingScrollIndex}
+                    />
+                )}
+            </div>
+
+            {/* Image Content */}
+            {screenshot && (
+                <div
+                    className="logviewer-image-div"
+                    onClick={() => setScreenshot(null)}
+                >
+                        <img
+                            src={screenshot}
+                            alt="screenshot"
+                            className="logviewer-image-content"
+                        />
+                </div>
+            )}
+        </div>
+    );
+}
+
+/**
+ * Handles row click events in the log viewer.
+ * 
+ * This function manages row selection with the following behaviors:
+ * - Ignores clicks on links within the row
```

Comment:
Just curious: does this mean that we can't click a link to view the raw inspect logs?